crates/rpc/rpc/src/trace.rs

```diff
@@ -343,6 +315,12 @@ where
             }
         }
 
+        // filter out traces that don't match the filter
+        all_traces = all_traces
+            .into_iter()
+            .filter(|trace| matcher.matches(&trace.trace))
+            .collect::<Vec<_>>();
```

Comment:
I think we need to be a bit more careful here now that we no longer prefilter blocks because this has now all the blocks in memory

perhaps we can apply the filter in the `trace_block_until(` call instead so we only buffer traces that we need?

also no longer sure about the `self.calculate_base_block_reward(` should we only do this for blocks that have at least one match?

Reply:
Yeah, inside the tracing should be more efficient. 

And for the block rewards, I think it's similar to normal traces, simple match should be sufficient. 

Btw, do we have any plans to indexing the trace from/to address, similar to erigon's TraceFromIndex and TraceToIndex?

Reply:
> Btw, do we have any plans to indexing the trace from/to address

nothing final yet, but we're considering it either with or without breaking the db, need to figure out if it's possible to introduce this without requiring a resync

perhaps you can open an issue for this, don't think this is tracked

Reply:
Thanks, if we are planning to implement it, I want to have a try